Adds getPortProperties for Windows

RomanBelkov opened the pull request:
As the main library does not accept pull requests and presumably cold dead, I pull request here, so the complete getPortProperties will be available more than in one repo.

I also request a code review, because my C++ skills are pretty bad.

RomanBelkov committed:
refactors getPortProperties, adds Win support to Java code

RomanBelkov committed:
adds Win cpp code with getPortProperties support

also made a few changes so that MS compiler would be able to compile

RomanBelkov committed:
multiple fixes for Win native getPortProperties

RomanBelkov committed:
adds new Win .dll's with getSerialPortProperties

Compiled in MSVC 2017.
Needs to add setupapi.lib for linker & give path to JDK includes.

RomanBelkov committed:
Adds 230400 baudrate to available rates

Tested on Win & Linux

RomanBelkov committed:
Merge branch 'baudrate_support'

gohai commented:
@RomanBelkov Thank you for this contribution! I am a bit busy with other things, but I will definitely have a close look soon.

RomanBelkov committed:
reads string instead of just part before the first space in LinuxPortProps

RomanBelkov committed:
Adds Windows provided by bus description to port properties

It appeared, that if you change device description (CP2102 in my case),
you'll still get default description given by Windows where the new, changed
device description is expected. Luckily, in Windows 7 and above there is a way to do so.

Note that this commit will probably break everything on XP & Vista.

RomanBelkov committed:
NULL is not a valid return from functions with wstring ret type
